src/zkevm_specs/evm/step.py

```diff
@@ -40,6 +40,10 @@ class StepState:
     memory_size: FQ
     state_write_counter: FQ
 
+    # log index of current block, this field maybe moved if we find them
+    # not often used.
+    log_index: int
```

Comment:
I understand `aux_data` fields are used as constant parameter,  `log_index` will change per any logs step(log0--log4), putting in step state we can use `constrain_step_state_transition` constrain its change, let me know if you also prefer that.

Reply:
But `aux_data` is defined as:
```python
# Auxilary witness data needed by gadgets
aux_data: Any
```

So either the comment on the code is wrong, or the values are not constant as this contains "witness data".

WDYT @DreamWuGit @ed255 ?

Reply:
oh,  I understood  you mean put `log_index` into `aux_data`,   them `aux_data` fields are constant witnesses, like   `CopyToMemoryAuxData`,  `CopyToLogAuxData:
    src_addr: FQ
    bytes_left: FQ
    src_addr_end: FQ`
```
class CopyToMemoryAuxData:
    src_addr: FQ
    dst_addr: FQ
    bytes_left: FQ
    src_addr_end: FQ
    from_tx: FQ
    tx_id: FQ.     
........
class CopyToLogAuxData:
    src_addr: FQ
    bytes_left: FQ
    src_addr_end: FQ
```. 
it will be hard to check if the 'log_index' change correctly when execution.

Reply:
Hmmm I see. 
No worries then. I was just trying to figure out if there was a way to not need new structures. Or fit them inside ones that we already use to reduce complexity :)

Reply:
@ed255 thanks  it can be a way, but so far I didn't come out how to exactly constrain it in the state circuit side, so put it in the step for now.  glad to hear more ideas!

Reply:
Well, maybe it's crazy but we could track it as an Operation. As we do with Stack/Memory/Storage read/write ops and add a section for it in the `rw_table`. 

On that way you should be able to access from both sites as the tables are the same.

WDYT?

Reply:
the original concern here is removing `log_index` from step, instead solely rely on state circuit to constrain validity by ed255,  I  would agree almost all what you said above :）

Reply:
```suggestion
    # log index of current block, this field maybe moved if we find them
    # not often used.
    log_id: FQ
```

Reply:
open an issue #141 to track state circuit part of work related to `Log` section!